src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -68,6 +74,34 @@ public void Unregister(Workspace workspace, bool blockingShutdown = false)
             {
                 source.Cancel();
             }
+
+            ClearDiagnostics(workspace);
```

Comment:
But what if someone just spins up a workspace for their own purpose.  Can they experience this leak?  I'm trying to figure out how we can come up with a pattern that prevents this issue for all consumers in the future as well.